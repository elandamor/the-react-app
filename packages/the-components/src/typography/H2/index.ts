import { space, styled, StyledSystemProps, typography } from '@app/theme';

/**
 * @render react
 * @name H2 component
 * @description Heading level 2.
 * @example
 *  <H2>Heading</H2>
 */

const H2 = styled.h2<StyledSystemProps>`
  ${space};
  ${typography};
`;

H2.defaultProps = {
  color: 'text.default',
  fontSize: [7, 8],
};

export default H2;
