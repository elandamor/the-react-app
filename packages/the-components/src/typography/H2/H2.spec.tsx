import { render } from '@testing-library/react';
import React from 'react';
import H2 from './index';

describe('H2', () => {
  it('renders without crashing', () => {
    const { container } = render(<H2>Heading</H2>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('h2');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct text', () => {
    const { getByText } = render(<H2>Heading</H2>);
    expect(getByText('Heading')).toBeTruthy();
  });
});
