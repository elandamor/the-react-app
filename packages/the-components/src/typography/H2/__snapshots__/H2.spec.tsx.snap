// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 renders without crashing 1`] = `
.emotion-0 {
  font-size: 64px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 72px;
  }
}

<h2
  class="emotion-0"
>
  Heading
</h2>
`;
